### DISCUSSION TITLE
Teaching R Markdown vs. Quarto?

### DISCUSSION THREAD

**Discussion by djvill** [quarto-dev/quarto-cli#1848] [Q&A]
I teach a graduate data science class that includes learning about reproducible research; we primarily use R in RStudio, and in the past I've used R Markdown both as a topic itself and as a teaching tool (e.g., for handouts). My average student is quite new to programming, needs a good deal of scaffolding, and will be a light-to-moderate R user. For a class starting this month, what are the pros and cons of teaching students about R Markdown vs. Quarto?

**Comment by cscheid** [comment]
We have an FAQ here https://quarto.org/docs/faq/rmarkdown.html

**Comment by djvill** [comment]
The FAQ doesn't address my question.

**Reply by mcanouil** [reply]
The FAQ contains the main questions and comparisons about formats which should help you decide.
In the end, you have to decide for yourself and make a choice which can evolve.

Because Quarto was release this summer, only workshops were given so far, thus I don't think you are likely to get "pros and cons of teaching students".

More contents with blog posts, workshos and tutorials: https://github.com/mcanouil/awesome-quarto#contents

**Comment by camilogarciabotero** [comment]
Hey,

I had a similar experience and found that Quarto is actually much better for beginners. The visual editor in Rstudio is a nice transition and there is a lot of autocompletion in the YAML that reduces the leap into dynamic documents. There is another [discussion](https://github.com/quarto-dev/quarto-cli/discussions/424) about the difference coming from Rmarkdown. I also found this [post](https://www.infoworld.com/article/3666743/what-is-quarto-rstudio-quietly-rolls-out-next-generation-r-markdown.html). I Hope this helps you!

**Comment by richardjtelford** [comment]
I've moved from teaching student how to use rmarkdown to quarto.

 For me the main advantages were

- not needing to have to worry about html_document vs bookdown::html_document2 for cross-references
- easy integration of python julia etc
-  future proofing

Only disadvantage I've had is needing to install quarto - some of my students don't have admin permission to do this.

My tutorial is at https://biostats-r.github.io/biostats/quarto/

**Comment by january3** [comment]
I am not planning to move from Rmarkdown to quarto in beginners classes, because this imposes an additional layer of complexity – installing quarto in addition to Rstudio and R, making sure that Rstudio version used supports quarto etc. With a hundred students, this would cause mayhem. It is bad as it is already, there is always 10% students who have trouble installing R/Rstudio as well as Rtools / developer tools and libraries (especially with Windows). I will wait another year at least with the transition.
